### PR TITLE
Rendering Assets and Entries

### PR DESCRIPTION
**Pull request by JuliusKarl**
**Loading Body Media:**

- Each content page loads a custom ngx-contentful-rich-text package and passes specified node data to a custom component >> Defer richTextToHTML

- Each content item stores its data in a singleton body-media.service.ts which is then called by the custom component rendering body media content
The custom component determines what kind of body media (entry/asset) was passed in and renders accordingly

- Asset and Entry hyperlinks are treated as standard hyperlinks

- Each Entry type is rendered in its own unique component

**Components to Review:**
Significant
- `body-media` in services >> Singleton service to store current body data in a `BehaviorSubject`
- `body-media` in components/shared >> Calls body data from `body-media.service.ts` and renders to page

Optional
- `article-entry` in components/shared
- `asset-block` in components/shared
- `cards` in components/shared
- `case-study` in components/shared
- `entry-block` in components/shared
- `entry-inline` in components/shared
- `equipment-entry` in components/shared
- `event-entry` in components/shared
- `large-cards` in components/shared
- `link-card` in components/shared
- `mini-cards` in components/shared
- `official-document` in components/shared
- `service` in components/shared
- `software` in components/shared
- `video-card` in components/shared
- ^ some of these aren't used at all but may in the future depending on rendering changes
- Any changes to existing front-end components are UI refactors which you can also review 

EDIT:

- I did a majority of refactoring on this branch hence the large number of file changes
- There may be a horizontal scrollbar bug that I can't replicate due to the banner width. It may or may not be an OS related problem but let me know if it shows up.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -275,7 +275,7 @@ pipeline {
         }
         failure {
             echo 'Jenkins job failed :('
-            slackSend(channel: slackChannel, tokenCredentialId: slackCredentials, color: "#FF9FA1", message: "🔥 Build failed - ${env.JOB_NAME} ${env.BUILD_NUMBER} (<${env.BUILD_URL}|Open>)")
+            // slackSend(channel: slackChannel, tokenCredentialId: slackCredentials, color: "#FF9FA1", message: "🔥 Build failed - ${env.JOB_NAME} ${env.BUILD_NUMBER} (<${env.BUILD_URL}|Open>)")
```
Comment (rosemcc) [inline Jenkinsfile:278]: uncomment

File: research-hub-web/package.json [hunk 2]
```diff
@@ -106,7 +106,8 @@
     "tslint": "~6.1.0",
     "typescript": "~4.0.5",
     "wait-on": "^5.2.0",
-    "webdriver-manager": "^12.1.7"
+    "webdriver-manager": "^12.1.7",
+    "ngx-contentful-rich-text": "^0.4.0"
```
Comment (rosemcc) [inline research-hub-web/package.json:110]: should ngx-contentful-rich-text be under dependencies rather than devDependencies?

**Comment by uoa-noel** [comment]
Vertical scrollbars in article pages as stated in https://github.com/UoA-eResearch/hub-stack/pull/56#issuecomment-768545535 are now fixed. Nice one! :+1:

**Comment by rosemcc** [comment]
Hey Julius, it's all looking good, nice job with the assets and entries rendering service. 
Just a couple of issues still with Jenkins and tests..

Unit tests are passing locally. But I had to increase the Chrome timeout in karma.conf.js (browserNoActivityTimeout: 60000, browserDisconnectTimeout: 60000) to get the tests to run. I guess the build is taking quite a while so it's timing out. Also having to set NODE_OPTIONS=--max_old_space_size=8192 to get it to actually build too. So still some optimisation to do around bundle sizes etc. 

e2e tests just had one test fail from the articles page: 1) ResearchHubs Article Pages
       can display a list of its keywords:
     AssertionError: Timed out retrying: Expected to find element: `#keywords`, but never found it.

Otherwise e2e seems good. 

Jenkins seems to be hanging still after the unit tests, so maybe again something to do with the bundle size?

**Pull request merged**